app/src/lib/stores/app-store.ts

```diff
@@ -6404,56 +6327,110 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return
     }
 
-    const { progress, targetBranchUndoSha } = snapshot
-
-    this.repositoryStateCache.updateCherryPickState(repository, () => ({
-      progress,
-      targetBranchUndoSha,
-    }))
+    this.repositoryStateCache.updateMultiCommitOperationState(
+      repository,
+      () => ({
+        progress: snapshot.progress,
+      })
+    )
   }
 
   /** display the cherry pick flow, if not already in this flow */
   private async showCherryPickConflictsDialog(
     repository: Repository,
-    conflictState: CherryPickConflictState
+    conflictState: CherryPickConflictState,
+    multiCommitOperationState: IMultiCommitOperationState | null,
+    tip: Tip
   ) {
-    const alreadyInFlow =
-      this.currentPopup !== null &&
-      this.currentPopup.type === PopupType.CherryPick
+    const snapshot = await getCherryPickSnapshot(repository)
 
-    if (alreadyInFlow) {
+    if (snapshot === null) {
+      log.error(
+        `[showCherryPickConflictsDialog] unable to get cherry-pick status from git, unable to continue`
+      )
       return
     }
 
+    if (multiCommitOperationState === null && tip.kind === TipState.Valid) {
+      // This is only true is we get here when opening the app and therefore we
+      // don't know some of this data
+      this._initializeMultiCommitOperation(
+        repository,
+        {
+          kind: MultiCommitOperationKind.CherryPick,
+          sourceBranch: null,
+          branchCreated: false,
+          commits: snapshot.commits,
+        },
+        tip.branch,
+        snapshot.commits,
+        null
+      )
+
+      this.repositoryStateCache.updateMultiCommitOperationUndoState(
+        repository,
+        () => ({
+          undoSha: snapshot.targetBranchUndoSha,
+          branchName: '',
+        })
+      )
+
+      this.repositoryStateCache.updateMultiCommitOperationState(
+        repository,
+        () => ({
+          progress: snapshot.progress,
+        })
+      )
+    }
+
+    // are we already in the conflicts flow?
+    const alreadyInFlow =
+      this.currentPopup !== null &&
+      this.currentPopup.type === PopupType.MultiCommitOperation &&
+      multiCommitOperationState !== null &&
+      (multiCommitOperationState.step.kind ===
+        MultiCommitOperationStepKind.ShowConflicts ||
+        multiCommitOperationState.step.kind ===
+          MultiCommitOperationStepKind.ConfirmAbort)
```

Comment:
Personally I prefer checking the contents of an array instead: `[MultiCommitOperationStepKind.ShowConflicts, MultiCommitOperationStepKind.ConfirmAbort].includes(multiCommitOperationState.step.kind)` (maybe extracting that array into a constant first)

What do you think?

Reply:
:D - I actually considered just making a helper function for this annoyingly long check since it is used twice. Once in cherry-pick method and once in merge method. However, there are a couple of methods that say "cherryPick" in them in the app-store and maybe dispatcher that I think can be refactored out because we are now using the multi-commit-operation logic and this is one of them. That is what my next PR will be doing. Ok with going ahead with this for now since this code may be removed anyways (if it doesn't I can address it in next PR)?

Reply:
SGTM!